pkg/util/http/helpers.go

```diff
@@ -77,3 +78,12 @@ func Put(ctx context.Context, URL string, headers map[string]string, body []byte
 
 	return parseResponse(res, "PUT", URL)
 }
+
+// SetJSONError writes a server error as JSON with the correct http error code
+func SetJSONError(w http.ResponseWriter, err error, errorCode int) {
+	body, _ := json.Marshal(map[string]string{"error": err.Error()})
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(errorCode)
+	fmt.Fprintln(w, string(body))
```

Comment:
I used what is used in [`http.Error()` function](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/net/http/server.go;l=2175).